### ISSUE TITLE
Damage Addition at Setup

### ISSUE THREAD

**Issue by tksolway** [Aatu/FieryVoid#482]
In order to support Campaign play, do you think it's possible to add a mechanism to the fleet builder to set damage levels on a ship?  For instance: Can we set the amount of damage that pre-exists on a system up to and including destroyed.  

As a secondary objective, perhaps we could add armour damage as well.

**Comment by Zero1627** [comment]
Yes, such mechanism would certainly be possible. Adding appropriate damage is relatively simple, in fact (such mechanism I use in self-destruct option, and Chris used in misile rack explosions). 

Main dificulty would be front end interface, where player could decide what damage he wants... 
Perhaps simplest would be specifying overall damage level for each section? With MCVs and LCVs around that would need more thought, too - so specifying desired damage is easy.

**Comment by tksolway** [comment]
We could probably make the interface pretty simple.  Left click to add damage and right click to remove it.

**Comment by Zero1627** [comment]
well, I don't feel nearly as good with front end as with back end - so it does not seem as simple to me.
Also, if You want to add point for point - You get hundred of clicks for any reasonably damaged ship... possible to do but lots of work!

**Comment by Zero1627** [comment]
>As a secondary objective, perhaps we could add armour damage as well.
Adding armour damage (and criticals in general) would be way more complicated (as previously, primarily in user interface layer)

**Comment by tksolway** [comment]
Ya, I’m not sure if it’s possible, but maybe we can have holding down CTRL for +5 or -5 on a click.

The other options would be to have small boxes next to each system to add numbers for the amount of damage.  Not sure how easy that would be to display though.

I’m not great with the interface either, but I can try and figure it out.

**Comment by melkorium** [comment]
I've been thinking about this off and on, I think there's two parts to it that could be fun.

The first is extending Saved Fleet functionality so that players can save a fleet during a live game, including the damage and (potentially) crits.  Then they should be able to load this damaged fleet in Fleet Selection.  Would require a new db table and buyingGamePhase would need to create the damage I guess, but fairly doable.

The second part would complement this by overhauling ship windows in gamelobby to use react elements so that user can click on systems and set their health (on bought ships only, but there is a marker for that now).  No idea how complex that would be, but could lead to other advantages.

**Comment by melkorium** [comment]
Completed 12.8.26
